.circleci/config.yml

```diff
@@ -61,11 +62,10 @@ jobs:
     parameters:
       python-version:
         type: string
-      xcode:
-        type: string
 
     macos:
-      xcode: << parameters.xcode >>
+      xcode: 15.3.0
```

Comment:
(minor) why remove the `xcode` parameter? Seems reasonable to have for future updates.

Reply:
We don't use this pattern elsewhere, and IMO it just moves the source of truth further from the relevant code.

Reply:
@arcondello, we do use this pattern in quite a few repos (dwave-cloud-client, dwave-system, dwave-hybrid, dwave-inspector, dwavebinarycsp, here, orb-ocean, orb-examples) and I introduced it primarily to centralize xcode definition (single source of truth) and re-use it for cache keys. Alternatively, on each xcode update one needs to remember to update it in 2 or 3 places in the file).

Reply:
Ok, IMO this is not really worth a PR to revert, but can do so if folks feel strongly.

Reply:
I agree, not worth a PR, but just wanted to clarify the benefits (and rationale) for doing it the previous way.

Reply:
In fact, the only repos where we're not using this pattern on are those built with cibuildwheel. :smiley:

Reply:
Really we ought to make a PR to CircleCI's Python orb addressing https://github.com/CircleCI-Public/python-orb/issues/69. Then just inherit from that to install python :shrug: rather than mucking about with caches and pyenv etc.

Reply:
I agree. Although we implemented `pyenv-install-python`, `nuget-install-python` and `brew-install-pyenv` in our `orb-ocean`, it would be nicer to have `python/install` command in circleci, just like it exists on GitHub Actions.